src/types/convert.rs

```diff
@@ -54,43 +56,47 @@ impl From<scene::InputId> for InputId {
     }
 }
 
-impl TryFrom<Scene> for scene::SceneSpec {
+impl TryFrom<UpdateScene> for scene::SceneSpec {
     type Error = TypeError;
 
-    fn try_from(scene: Scene) -> Result<Self, Self::Error> {
-        fn from_output(output: Output) -> scene::OutputSpec {
-            scene::OutputSpec {
-                input_pad: output.input_pad.into(),
-                output_id: output.output_id.into(),
+    // very temporary and inefficient
+    fn try_from(update_scene: UpdateScene) -> Result<Self, Self::Error> {
```

Comment:
converts tree structure into a set of nodes, temporary until we add additional layer between rendering pipeline and API